daemon/container_unix.go

```diff
@@ -1025,11 +1033,11 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		container.updateSandboxNetworkSettings(sb)
 	}
 
-	if err := ep.Join(sb); err != nil {
+	if err = ep.Join(sb); err != nil {
 		return err
 	}
```

Comment:
we don't need to. The `container.cleanup` defer call in `container.Start` will take care of it by calling `releaseNetwork`